### PR TITLE
fix: MD046/code-block-style

### PR DESCRIPTION
**Pull request by nschonni**
Code block style of fenced to match nodejs style

### CODE REVIEW

File: locale/en/blog/feature/streams2.md
Comment (Trott) [inline locale/en/blog/feature/streams2.md]: This isn't code.

Comment (nschonni) [reply locale/en/blog/feature/streams2.md]: I agree, but it's currently rendered as a code block because of the indentation, so I didn't want to change the rendering

**Comment by XhmikosR** [comment]
@Trott please don't merge this!

@nschonni did you run `npm start` locally? This throws, or at least should throw, when a prism lang is specified that doesn't exist.

For example:

```
Failed to load prism syntax: console
Error: Cannot find module 'prismjs/components/prism-console.js'
Require stack:
- C:\Users\xmr\Desktop\nodejs.org\node_modules\metalsmith-prism\lib\index.js
- C:\Users\xmr\Desktop\nodejs.org\build.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:772:15)
    at Function.Module._load (internal/modules/cjs/loader.js:677:27)
    at Module.require (internal/modules/cjs/loader.js:830:19)
    at require (internal/modules/cjs/helpers.js:68:18)
    at requireLanguage (C:\Users\xmr\Desktop\nodejs.org\node_modules\metalsmith-prism\lib\index.js:42:11)
    at Object.<anonymous> (C:\Users\xmr\Desktop\nodejs.org\node_modules\metalsmith-prism\lib\index.js:86:11)
    at initialize.exports.each (C:\Users\xmr\Desktop\nodejs.org\node_modules\cheerio\lib\api\traversing.js:300:24)
    at C:\Users\xmr\Desktop\nodejs.org\node_modules\metalsmith-prism\lib\index.js:62:17
    at C:\Users\xmr\Desktop\nodejs.org\node_modules\lodash\lodash.js:4905:15
    at baseForOwn (C:\Users\xmr\Desktop\nodejs.org\node_modules\lodash\lodash.js:2990:24) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\xmr\\Desktop\\nodejs.org\\node_modules\\metalsmith-prism\\lib\\index.js',
    'C:\\Users\\xmr\\Desktop\\nodejs.org\\build.js'
  ]
}
```

**Comment by XhmikosR** [comment]
We really need #2539, although to be fair, in this case these are wrapped in try/catch so it doesn't throw. But at least we would see them in our console.

**Comment by nschonni** [comment]
OK, just removed the "console" ones with plain blocks

File: locale/en/knowledge/advanced/buffers/how-to-use-buffers.md
Comment (XhmikosR) [inline locale/en/knowledge/advanced/buffers/how-to-use-buffers.md:65]: Can you try with `bash` in such cases? Is it highlighted properly?

Comment (XhmikosR) [reply locale/en/knowledge/advanced/buffers/how-to-use-buffers.md:65]: Actually scratch that, this should happen in another PR. Just make sure any languages you specify exist in prism.

Comment (nschonni) [reply locale/en/knowledge/advanced/buffers/how-to-use-buffers.md:65]: No, bash wouldn't work since this is the Node REPL

**Comment by XhmikosR** [comment]
@Trott: do you know which library is used upstream for highlighting? It seems [metalsmith-prism](https://github.com/Availity/metalsmith-prism) is archived, so it might be worth looking into switching to the same way it's done upstream in another PR.

File: locale/en/blog/feature/streams2.md
Comment (nschonni) [reply locale/en/blog/feature/streams2.md]: Found that the main docs were using blockquotes for these, so I swapped that

**Comment by Trott** [comment]
> @Trott: do you know which library is used upstream for highlighting? It seems [metalsmith-prism](https://github.com/Availity/metalsmith-prism) is archived, so it might be worth looking into switching to the same way it's done upstream in another PR.

Looking at https://github.com/nodejs/node/blob/6723169097759502da11c47389ffb3d650404417/tools/doc/html.js, I guess we're using https://github.com/remarkjs/remark-rehype.

**Comment by XhmikosR** [comment]
This isn't ready IMO.

When we use fenced code blocks, we need to remove any indented code otherwise we'll end up with extra spaces in the snippets. It seems there are some cases this is missed like https://github.com/nodejs/nodejs.org/pull/2557/files#diff-74f34b44a0285dc0c6cd1f2eeed2e2b5

I didn't go through all of the changes, so I don't know if there are more.

File: locale/pt-br/knowledge/errors/what-are-the-error-conventions.md
Comment (XhmikosR) [inline locale/pt-br/knowledge/errors/what-are-the-error-conventions.md]: These comments will show up in the code block

**Comment by XhmikosR** [comment]
LGTM except for my comment above

File: locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md
Comment (XhmikosR) [inline locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md]: Maybe change this to 2-space indentation. There might be more cases like this

File: locale/pt-br/knowledge/errors/what-are-the-error-conventions.md
Comment (XhmikosR) [inline locale/pt-br/knowledge/errors/what-are-the-error-conventions.md]: Please add the missing 2 `)`

**Comment by nschonni** [comment]
@Trott I think all the major stuff is resolved now. The indentation stuff is as-is for the existing examples

**Comment by Trott** [comment]
> @Trott I think all the major stuff is resolved now. The indentation stuff is as-is for the existing examples

This is a big change set. @nodejs/website Anyone up for trying to give this a review?

File: locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md
Comment (XhmikosR) [reply locale/en/knowledge/getting-started/control-flow/how-to-write-asynchronous-code.md]: This still has 4-space indentation

**Comment by XhmikosR** [comment]
@Trott we should move forward with #2523 and #2524 so that we don't have to review the snippets manually and that we are consistent. I've left a comment on #2523.

As far as this PR goes, it's definitely an improvement. I'd only go through the snippets and fix the indentation while at it. Assuming we don't use any code block languages that prism doesn't support (like `console`) this should be OK. It also fixes a broken example in the pt_BR translation.

**Comment by nschonni** [comment]
Easiest way to review it is with the "whitespace ignored" option set, then you just see the code fence lines and the example fix

**Comment by XhmikosR** [comment]
Yeah, but in this case whitespace does matter. So, the only safe way is viewing the rich preview of each file which takes time though.

File: locale/en/knowledge/intermediate/how-to-log.md
Comment (XhmikosR) [inline locale/en/knowledge/intermediate/how-to-log.md:97]: This (in all languages) could be `bash`

File: locale/en/knowledge/getting-started/control-flow/what-are-callbacks.md
Comment (XhmikosR) [inline locale/en/knowledge/getting-started/control-flow/what-are-callbacks.md]: This still has 4-space indentation

**Comment by XhmikosR** [comment]
This looks good to me. There are 3 unresolved comments and then it can be merged IMO.

File: locale/en/knowledge/intermediate/how-to-log.md
Comment (XhmikosR) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: Maybe just leave it as is.

**Review by alexandrtovmach — APPROVED** [review]

File: locale/en/knowledge/intermediate/how-to-log.md
Comment (Trott) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: Yeah, please don't use <code>```bash</code> for this!

Use that for bash scripts only. It does confusing highlighting for command-line shell stuff. Use <code>```console</code> for that instead if you really want to specify a language/formatter/highlighter.

Comment (XhmikosR) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: @Trott `console` is not a valid prism language thus this will fail.

Comment (nschonni) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: Yeah, we figured out that "console" chokes out metalsmith, unless some of the dependency updates done recently fix that

Comment (XhmikosR) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: See https://github.com/nodejs/nodejs.org/pull/2557#issuecomment-529292779

Comment (Trott) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: Ah, OK, yeah, in that case, yes, specifying nothing is 👍.  Sorry I'm having difficulty keeping up.

Comment (XhmikosR) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: Nope, this is how metalsmith-prism works. If we want to change this, we need to fork the project.

Comment (XhmikosR) [reply locale/en/knowledge/intermediate/how-to-log.md:97]: which has been archived BTW

**Comment by XhmikosR** [comment]
@nschonni can you just address the 2 indentation comments while you change the code snippets?

Otherwise someone with push rights could fix them and then just squash.

**Pull request merged**
